vpr/src/route/rr_graph_writer.cpp

```diff
@@ -37,6 +39,15 @@ void write_rr_segments(fstream& fp, const std::vector<t_segment_inf>& segment_in
 /* This function is used to write the rr_graph into xml format into a a file with name: file_name */
 void write_rr_graph(const char* file_name, const std::vector<t_segment_inf>& segment_inf) {
     fstream fp;
+    FILE* fb;
+    std::string filename_str(file_name);
+    bool binary_mode = (filename_str.substr(filename_str.length() - 7) == "bin.xml");
+    std::string bin_file_name;
+    if (binary_mode) {
+        bin_file_name = filename_str.substr(0, filename_str.length() - 4);
+        printf("bin file |%s|\n", bin_file_name.c_str());
```

Comment:
I suppose this is a log print. If I am correct IMO it would be best to use the VTR built-in logging mechanism:
```
VTR_LOG("RR_graph binary file: |%s|\n", bin_file_name.c_str());
```